src/lib/operators/join_nested_loop_a.cpp

```diff
@@ -292,12 +326,17 @@ class JoinNestedLoopA::JoinNestedLoopAImpl : public AbstractJoinOperatorImpl {
       for (const ChunkOffset& offset_in_left_value_column : *(chunk_offsets_in_left)) {
         ChunkOffset row_right = 0;
         for (const ChunkOffset& offset_in_right_value_column : *(chunk_offsets_in_right)) {
+          auto current_right = RowID{context->chunk_id_right, row_right};
+          auto current_left = RowID{context->chunk_id_left, row_left};
+
+          // Check null values
+          auto comparing_null =
+              is_left_value_null(offset_in_left_value_column) || is_right_value_null(offset_in_right_value_column);
+
```

Comment:
if this is `true`, we do not need to the comparison below, right? maybe wrap that around an `if` statement

Reply:
yes, that is true. but adding another `if` check in there makes the code look a little weird. We would be doing:
    
    bool is_match = false;
    auto comparing_null = left_null... || right_null...;
    if (!comparing_null) {
      is_match = compare()
    }

    if (!comparing_null || is_match) { ... }
    
Same check twice or we add another level of indentation. I would leave it as is because the compare function is rather cheap to call just for the sake of better readability.

Reply:
Maybe I'm missing something but would something like this work?

```C++

// first, check for a non-match because a NULL value is involved
auto is_match = 
  !(is_left_value_null(offset_in_left_value_column) || is_right_value_null(offset_in_right_value_column));

if (is_match) {
  // second, check for a match by value
  is_match = context->compare_func(get_left_column_value(offset_in_left_value_column),
                                   get_right_column_value(offset_in_right_value_column));
}

if (is_match) {
  // For outer joins we need to mark these rows, since they don't need to be added later on.
  if (context->join_mode == JoinMode::Right || context->join_mode == JoinMode::Outer) {
    (*unmatched_rows_map_right)[current_right] = false;
  }
  if (context->join_mode == JoinMode::Left || context->join_mode == JoinMode::Outer) {
    (*unmatched_rows_map_left)[current_left] = false;
  }
  write_poslists(context, row_left, row_right);
}
```

Reply:
yes, that would work. will change